src/gtest/test_foundersreward.cpp

```diff
@@ -107,11 +107,18 @@ TEST(founders_reward_test, mainnet) {
     CChainParams params = Params();
 
     int maxHeight = params.GetConsensus().GetLastFoundersRewardBlockHeight();
-    std::set<std::string> addresses;
-    for (int i=1; i<=maxHeight; i++) {
-        addresses.insert(params.GetFoundersRewardAddress(i));
-    }
-    ASSERT_TRUE(addresses.size()==NUM_MAINNET_FOUNDER_ADDRESSES);
+    checkNumberOfUniqueAddresses(maxHeight, NUM_MAINNET_FOUNDER_ADDRESSES);
+}
+
+
+#define NUM_TESTNET_FOUNDER_ADDRESSES 48
+
+TEST(founders_reward_test, testnet) {
+    SelectParams(CBaseChainParams::TESTNET);
+    CChainParams params = Params();
+
+    int maxHeight = params.GetConsensus().GetLastFoundersRewardBlockHeight();
+    checkNumberOfUniqueAddresses(maxHeight, NUM_TESTNET_FOUNDER_ADDRESSES);
```

Comment:
Since ``checkNumberOfUniqueAddresses`` depends on the global ``Params()`` anyway, might as well compute ``maxHeight`` there rather than passing it in.